react/src/SearchVariants/components/SearchVariantsPage.tsx

```diff
@@ -20,12 +21,44 @@ const useStyles = makeStyles(theme => ({
 }));
 
 const SearchVariantsPage: React.FC<SearchVariantsPageProps> = () => {
-    const [selectedGene, setSelectedGene] = useState<Gene>();
+    const [selectedGenes, setSelectedGenes] = useState<string[]>([]);
+    const [error, setError] = useState(false);
 
-    console.log(selectedGene);
-    // currently a dummy endpoint
-    /* eslint-disable @typescript-eslint/no-unused-vars */
-    //const { data: variants } = useVariantsQuery({ search: selectedGene }, "csv", !!selectedGene);
+    const { enqueueSnackbar } = useSnackbar();
+
+    const updateSelectedGenes = (gene: string) => {
+        if (gene && !selectedGenes.includes(gene)) {
+            setSelectedGenes(selectedGenes.concat(gene));
+        } else {
+            setSelectedGenes(selectedGenes.filter(g => g !== gene));
+        }
+    };
+
+    const { data: blob, refetch } = useVariantsQuery({ panel: selectedGenes.join(";") }, "csv", {
+        enabled: false,
+        onError: response => {
+            if (response.status === 400) {
+                setError(true);
+            } else {
+                const errorText = response.statusText;
+                enqueueSnackbar(`Query Failed. Error: ${errorText}`, { variant: "error" });
+            }
+        },
+    });
+
+    const clearError = () => {
+        if (error) {
+            setError(false);
+            setSelectedGenes([]);
+        }
+    };
+
+    useEffect(() => {
+        if (blob && selectedGenes) {
+            downloadCsvResponse(blob);
+            setSelectedGenes([]);
```

Comment:
I don't think we should be wiping out the user's selection on download, but otherwise this seems good as a basis.

Reply:
updated in [6362962](https://github.com/ccmbioinfo/stager/pull/518/commits/636296232ef268ace401e3890a4b8ecd76d774bc)